app/scripts/controllers/app-state-controller.ts

```diff
@@ -0,0 +1,860 @@
+  handler: () => AppStateControllerState;
+};
+
+/**
+ * Actions exposed by the {@link AppStateController}.
+ */
+export type AppStateControllerActions = AppStateControllerGetStateAction;
+
+/**
+ * Actions that this controller is allowed to call.
+ */
+export type AllowedActions = AddApprovalRequest | AcceptRequest;
+
+/**
+ * Event emitted when the state of the {@link AppStateController} changes.
+ */
+export type AppStateControllerStateChangeEvent = {
+  type: 'AppStateController:stateChange';
+  payload: [AppStateControllerState, []];
+};
+
+/**
+ * Events emitted by {@link AppStateController}.
+ */
+export type AppStateControllerEvents = AppStateControllerStateChangeEvent;
+
+/**
+ * Events that this controller is allowed to subscribe.
+ */
+export type AllowedEvents =
+  | PreferencesControllerStateChangeEvent
+  | KeyringControllerQRKeyringStateChangeEvent;
+
+export type AppStateControllerMessenger = RestrictedControllerMessenger<
+  typeof controllerName,
+  AppStateControllerActions | AllowedActions,
+  AppStateControllerEvents | AllowedEvents,
+  AllowedActions['type'],
+  AllowedEvents['type']
+>;
+
+type PollingTokenType =
+  | 'popupGasPollTokens'
+  | 'notificationGasPollTokens'
+  | 'fullScreenGasPollTokens';
+
+type SecurityAlertResponse = {
+  block?: number;
+  description?: string;
+  features?: string[];
+  providerRequestsCount?: Record<string, number>;
+  reason: string;
+  result_type: string;
+  securityAlertId?: string;
+  source?: SecurityAlertSource;
+};
+
+type AppStateControllerOptions = {
+  addUnlockListener: (callback: () => void) => void;
+  isUnlocked: () => boolean;
+  initState?: Partial<AppStateControllerState>;
+  onInactiveTimeout?: () => void;
+  // TODO: Remove this as soon as PreferencesController upgrade to BaseControllerV2 merges with develop
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  preferencesStore: any;
+  messenger: AppStateControllerMessenger;
+  // TODO: Replace `any` with type
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  extension: any;
+};
+
+const getInitState = (
+  initState?: Partial<AppStateControllerState>,
+): AppStateControllerState => ({
+  timeoutMinutes: DEFAULT_AUTO_LOCK_TIME_LIMIT,
+  connectedStatusPopoverHasBeenShown: true,
+  defaultHomeActiveTabName: null,
+  browserEnvironment: {},
+  popupGasPollTokens: [],
+  notificationGasPollTokens: [],
+  fullScreenGasPollTokens: [],
+  recoveryPhraseReminderHasBeenShown: false,
+  recoveryPhraseReminderLastShown: new Date().getTime(),
+  outdatedBrowserWarningLastShown: null,
+  nftsDetectionNoticeDismissed: false,
+  showTestnetMessageInDropdown: true,
+  showBetaHeader: isBeta(),
+  showPermissionsTour: true,
+  showNetworkBanner: true,
+  showAccountBanner: true,
+  trezorModel: null,
+  onboardingDate: null,
+  newPrivacyPolicyToastClickedOrClosed: null,
+  newPrivacyPolicyToastShownDate: null,
+  hadAdvancedGasFeesSetPriorToMigration92_3: false,
+  ...initState,
```

Comment:
hmm, I'm aware that this was already here, but do you think we could move this to the end of the object?

Alternatively, do you know why this was placed here?

Reply:
I believe it's intentional, as we want to ensure that these state properties are properly reset. For instance, this explains why `signatureSecurityAlertResponses` is placed at the bottom: https://github.com/MetaMask/metamask-extension/pull/22778/files

Reply:
Hmm, I see. But then do you think that the `initState?: Partial<AppStateControllerState>` type might be misleading? Some of the keys passed as `initState` would be simply ignored

Reply:
You raise a valid point; we should exclude these properties to prevent any confusion I think.